src/hooks/useSavingsAccount.js

```diff
@@ -1,49 +1,198 @@
 
-  let tokens = accountTokens;
+  const { accountAddress, network } = useAccountSettings();
 
-  const shouldAddDai =
-    includeDefaultDai && !accountHasCDAI && !isEmpty(daiMarketData);
-  if (shouldAddDai) {
-    tokens = concat(accountTokens, {
-      ...daiMarketData,
-    });
-  }
+  const hasAccountAddress = !!accountAddress;
+  const { shouldRefetchSavings } = useSelector(
+    ({ data: { shouldRefetchSavings } }) => ({
+      shouldRefetchSavings,
+    })
+  );
 
-  useEffect(() => {
-    if (justBecameActive) {
-      dispatch(savingsLoadState());
+  const { data, error, loading, refetch: refetchSavings } = useQuery(
+    COMPOUND_ACCOUNT_AND_MARKET_QUERY,
+    {
+      client: compoundClient,
+      pollInterval: COMPOUND_QUERY_INTERVAL,
+      skip: !hasAccountAddress,
+      variables: { id: toLower(accountAddress) },
     }
-  }, [dispatch, justBecameActive]);
+  );
+
+  useEffect(() => {
+    if (!hasAccountAddress) return;
+    const fetchBackupSavings = async () => {
+      const backup = await getSavings(accountAddress, network);
+      setBackupAccountTokens(backup);
+    };
+    fetchBackupSavings();
+  }, [accountAddress, hasAccountAddress, network]);
 
-  // Reload on address change
   useEffect(() => {
-    if (previousAccount && previousAccount !== accountAddress) {
-      dispatch(savingsLoadState());
+    if (!hasAccountAddress || isNil(backupAccountTokens)) return;
+    const parseSavingsResult = async data => {
+      if (error) return;
+      let savingsAccountData = [];
+      const markets = keyBy(data?.markets || [], property('id'));
+
+      if (data) {
+        const resultTokens = data?.account?.tokens || [];
+
+        const parsedAccountTokens = resultTokens.map(token => {
+          const [cTokenAddress] = token.id.split('-');
+          const marketData = markets[cTokenAddress] || {};
+
+          const {
+            cToken,
+            exchangeRate,
+            supplyRate,
+            underlying,
+            underlyingPrice,
+          } = getMarketData(marketData);
+
+          const ethPrice = multiply(
+            underlyingPrice,
+            token.supplyBalanceUnderlying
+          );
+
+          const {
+            cTokenBalance,
+            lifetimeSupplyInterestAccrued,
+            supplyBalanceUnderlying,
+          } = token;
+
+          return {
+            cToken,
+            cTokenBalance,
+            ethPrice,
+            exchangeRate,
+            lifetimeSupplyInterestAccrued,
+            supplyBalanceUnderlying,
+            supplyRate,
+            type: assetTypes.cToken,
+            underlying,
+            underlyingPrice,
+          };
+        });
+        savingsAccountData = orderBy(
+          parsedAccountTokens,
+          ['ethPrice'],
+          ['desc']
+        );
+      } else if (loading) {
+        savingsAccountData = backupAccountTokens;
+      }
+
+      saveSavings(savingsAccountData, accountAddress, network);
```

Comment:
This should me moved inside the if cause there's no reason to read and write the same data to localstorage